File: Session17/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hm = new HashMap<Integer,Integer>();
        int n=nums.length;
        for(int i=0;i<n;i++)
        {
            int rem=target-nums[i];
            if(hm.containsKey(rem))
            {
                return new int[]{i,hm.get(rem)};
            }
            else
            {
                hm.put(nums[i],i);
            }
        }
        return new int[]{-1,-1};
    }
}

File: Session17/1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dicti={}
        for i in range(len(nums)):
            if target-nums[i] in dicti:
                return [dicti[target-nums[i]],i]
            if nums[i] not in dicti:
                dicti[nums[i]]=1